Add option to disable particle items

thexmanxyz opened the pull request:
I added an additional field to all relevant particles which contain a `collection.list` to allow that a user can selectively disable certain front-end items. The idea for this adaption comes from @yellowwebmonkey. The particles which I found and come into question are:

* Social
* Sample (Hydrogen)
* Content Cubes
* Horizontal Menu
* Content Tabs

Please also see the PR #2286 which contains the same functionality for the Owl Carousel.

thexmanxyz commented:
Tagging people @JoomFX @w00fz @n8solutions @yellowwebmonkey

JoomFX commented:
@thexmanxyz This is a great feature but in my opinion it should be implemented on the framework level somehow.
I'm simply not interested in modifying 70+ particles to add this feature :)

thexmanxyz commented:
@JoomFX Yeah that is also what I pointed out ;). A ideal solution would apply this automatically and handle the filtering of disabled ones.

mahagr commented:
@thexmanxyz I just have a single major wish: Please change the option to be positive instead of negative. This is for usability as it is better to keep enabled (green) state on left and disabled (red) on the right. :)

thexmanxyz commented:
@mahagr  I'm not sure what you mean because

```
.disabled:
  type: input.checkbox
  **label: Disabled**
  description: Enables or disables this item.
  **default: false**
```

The original commit was the other way around **Enabled Label** / `default: true` but obviously that won't work for `collection.list` items because they would not reconfigure automatically to default state. What that means is, that already existing items would be disabled until each list item is saved again. That's the reason I choose it this way. It's the same as in #2286 

Also there is not red and green because it is a default checkbox.

thexmanxyz commented:
@mahagr I updated the particles. I changed the description, the type is now `enable.enable` as you requested. I also changed the label to Enabled and on by default. However the issue from above is still there as new controls don't hold the default state until re-saved. But I now simply forced `|default(true)` on the control value which solves it for me. Now everything should be as requested.

mahagr commented:
Yes, you need to enter a default value in order to make it work before saving. That is fully expected as otherwise the default is null (not set).

thexmanxyz commented:
@mahagr I'm not sure if we are talking about the same. If I set up the following control:

```
        .enabled:
          type: enable.enable
          label: Enabled
          description: Enables the item on the front end.
          default: true
```
in a `collection.list`. The default state for `.enabled` (`true`) is only set if I open each already existing item and save it. Otherwise the state would be null (not set). That is what I experienced. Just reiterating the issue.

mahagr commented:
Yes, we are. What you're talking about is the default behaviour for configuration. If the value is not set, Gantry thinks it is null even if you had the default set in the blueprints. Blueprints are only used for displaying the form and saving it.

thexmanxyz commented:
@mahagr Alright, I'm glad that we talking about the same thing. Ok I now understand how the default values work! Thanks.
